tf2_ros/src/tf2_ros/buffer.py

```diff
@@ -146,11 +148,11 @@ def can_transform_full(self, target_frame, target_time, source_frame, source_tim
         """
         if timeout != rospy.Duration(0.0):
             start_time = rospy.Time.now()
-            r= rospy.Rate(20)
+            polling_interval = timeout * self.CAN_TRANSFORM_POLLING_SCALE
             while (rospy.Time.now() < start_time + timeout and 
                    not self.can_transform_full_core(target_frame, target_time, source_frame, source_time, fixed_frame)[0] and
                    (rospy.Time.now()+rospy.Duration(3.0)) >= start_time): # big jumps in time are likely bag loops, so break for them
-                r.sleep()
+                rospy.sleep(polling_interval)
```

Comment:
You imply that this is lower overhead, but you're calling rospy.sleep which is imported from rospy.timer.sleep which looks to me to be the same implementation that the rospy.timer.Rate.sleep() calls.

Reply:
I think the overhead is contributed by calling [_remaining()](https://github.com/ros/ros_comm/blob/08fcc096f4e822b56bfb5a30b0a09ec45f0802aa/clients/rospy/src/rospy/timer.py#L103) in rospy.timer.Rate.sleep() and creating a rospy.timer.Rate object per can_transform() call. While this overhead could be ignored in most cases, calling can_transform() frequently with a small timeout would found noticeable difference.

Reply:
ok, that makes sense. And we don't need this polling to be "precise"